### ISSUE TITLE
Big difference in feature importances between xgboost version

### ISSUE THREAD

**Issue by mchav** [dmlc/xgboost#11491]
When I train a model on xgboost 1.7 the importance gain of feature x is 5700 but on the latest version (3.0.2) it's 6400. And for feature y on xgboost 1.7 it has an importance gain of 844 but in the latest xgboost it has a gain of 13k.

For context the 1.7 hyper parameters are:

```
HYPERPARAMETERS = {
    'alpha': 9,
    'colsample_bylevel': 0.9,
    'colsample_bynode': 0.6,
    'colsample_bytree': 0.75,
    'early_stopping_rounds': 100,
    'eta': 0.05,
    'eval_metric': "aucpr",
    'gamma': 0.75,
    'lambda': 6,
    'max_depth': 6,
    'min_child_weight': 5,
    'num_round': 5000,
    'objective': "binary:logistic",
    'scale_pos_weight': 10,
    'seed': 1234,
    'subsample': 0.8,
    'tree_method': 'hist',
    'verbosity': 0,
}
```

And the latest parameters are:

```
xgb_params = {
              'eval_metric': 'aucpr',
              'early_stopping_rounds': 100,
              'num_workers': 256,
              'tree_method': 'hist',
              'n_estimators': 5000,
              'max_depth': 6,
              'min_child_weight': 5,
              'scale_pos_weight': 10,
              'gamma': 0.75,
              'learning_rate': 0.05,
              'subsample': 0.8,
              'colsample_bytree': 0.75,
              'colsample_bylevel': 0.9,
              'colsample_bynode': 0.6,
              'reg_alpha': 9,
              'reg_lambda': 6,
              'random_seed': 1234}
```

They are practically identical.

**Comment by mchav** [comment]
Setting `base_score = mean(label)` seems to address a large chunk of the problem.

**Comment by astariul** [comment]
@trivialfis In my training script, I'm loosing a few points of accuracy when updating from `2.1.4` to `3.0.2`.

I was creating my model like so : `m1 = xgb.XGBClassifier(max_depth=6, min_child_weight=1, gamma=0)`

I tried specifying the `tree_method` as you mentioned the default changed : `m1 = xgb.XGBClassifier(tree_method="approx", max_depth=6, min_child_weight=1, gamma=0)`

But I get the same result.
**Is there other default parameters that changed ?**

**Comment by trivialfis** [comment]
Is this multi class or binary? If it's binary, then you can change the base score to 0.5

**Comment by astariul** [comment]
It's binary classification. ~I tried to add base_score to 0.5, but I get the same score..~

@trivialfis Nevermind, it works, now I get the lower score both on `3.0.2` and `2.1.4` 😅 

I guess I'll try different values for `base_score` until I get back my good accuracy :)

**Comment by mchav** [comment]
@astariul idk if there is any randomness involved but does setting the same random seed help?
